examples/platforms/simulation/misc.c

```diff
@@ -113,3 +113,12 @@ otPlatMcuPowerState otPlatGetMcuPowerState(otInstance *aInstance)
 #if OPENTHREAD_CONFIG_PLATFORM_LOG_CRASH_DUMP_ENABLE
 otError otPlatLogCrashDump(void) { return OT_ERROR_NONE; }
 #endif
+
+void otPlatAssertFail(const char *aFilename, int aLineNumber)
+{
+    otLogCritPlat("assert failed at %s:%d", aFilename, aLineNumber);
+
+    // For debug build, use assert to generate a core dump
+    assert(false);
+    exit(1);
```

Comment:
Is `exit` needed after `assert`?

Reply:
The `assert()` does nothing if the ` NDEBUG` is defined. We need to call `exit()` in this case.

> If the macro NDEBUG is defined at the moment <assert.h> was last included, the macro assert() generates no code, and hence does nothing at all.

Refer to https://man7.org/linux/man-pages/man3/assert.3.html